### ISSUE TITLE
flycheck-javascript-eslint-executable nil but executable found.

### ISSUE THREAD

**Issue by PinkFLoyd92** [flycheck/flycheck#1492]
I am checking why the variable **flycheck-javascript-eslint-executable** is nil, if the config is found I thought it should be set.

The eslint and its config is found.
![image](https://user-images.githubusercontent.com/8844096/43962935-92d6fdac-9c7e-11e8-86d5-3fd0fb664c1a.png)

But the value of its executable is nil.
![image](https://user-images.githubusercontent.com/8844096/43962983-bab036ea-9c7e-11e8-8f3c-6b38871004b7.png)

How can I fix this?

**Comment by fmdkdd** [comment]
There is some confusion here.  `flycheck-javascript-eslint-executable` is a variable defined automatically by Flycheck to allow you to override the PATH environment.  See [the manual](http://www.flycheck.org/en/latest/user/syntax-checkers.html#change-syntax-checker-executables).

As the docstring states, `nil` is the default value, in which case Flycheck will try to locate `eslint` using your PATH.  It looks like it found eslint in your case.  If your `PATH` was not correctly setup, then you could have modified `flycheck-javascript-eslint-executable` to point to `eslint` directly.

In other words, there is nothing to fix.  Or have I misunderstood?
